src/firefly/java/edu/caltech/ipac/util/download/ConcurrentDownload.java

```diff
@@ -215,16 +210,53 @@ private static void logNotModified(File outfile, URL url) {
         log.info(send+file);
     }
 
-    private static void logSuccess(HttpResultInfo r, File outfile, URL url,  double dSeconds, int parts) {
+    private static void logSuccess(HttpResultInfo r, File outfile, URL url,  double dSeconds, int parts, HttpResultInfo headerR) {
         String formatedSize= FileUtil.getSizeAsString(r.getContentLength());
-        String send= String.format( "Download (%.1f sec, %s, %d parts): %s\n", dSeconds, formatedSize, parts, url.toString() );
-        String stat= String.format(
-                "        length: %d, contentType: %s, encoding: %s, disposition %s\n",
-                r.getContentLength(), r.getContentType(), r.getContentEncoding(), r.getContentDisposition() );
-        String file= "        File: "+ outfile.toPath();
-        log.info(send+stat+file);
+        String lastMod= r.getAttribute("Last-Modified")!=null ? ", Last-Modified: " +r.getAttribute("Last-Modified") : "";
+        log.info(
+                String.format( "DOWNLOAD (%.1f sec, %s, %d parts): Content-Type: %s, Content-Length: %s%s",
+                        dSeconds, formatedSize, parts, r.getContentType(), r.getContentLength(), lastMod),
+                "url:  "+ url.toString(),
+                "file: "+ outfile.toPath(),
+                "headers sent: " + sendHeadersToStr(headerR),
+                "more headers: "+otherHeadersToStr(r)
+        );
     }
 
+    private static String otherHeadersToStr(HttpResultInfo r) {
+        StringBuilder out= new StringBuilder();
+        int cnt=0;
+        for(var a : r.getAttributes().entrySet()) {
+            var k = a.getKey();;
+            if (r.isReservedKey(k) && !k.equalsIgnoreCase("<none>") &&
+                    !k.equalsIgnoreCase("content-type") && !k.equalsIgnoreCase("content-length") &&
+                    !k.equalsIgnoreCase("Last-Modified") ) {
```

Comment:
I spent some time studying this...should it be 

`if ( !r.isReservedKey(k) `... followed by the other conditions, instead?